src/bdr_solrizer/solrdocbuilder.py

```diff
@@ -70,15 +71,18 @@ def __init__(self, pid):
 
     def _get_files_response(self):
         url = f'{STORAGE_SERVICE_ROOT}{self.pid}/files/?{STORAGE_SERVICE_PARAM}&objectTimestamps=true&fields=state,size,mimetype,checksum,lastModified'
-        r = requests.get(url)
-        if r.status_code == 404:
-            raise ObjectNotFound()
-        if r.status_code == 410:
-            raise ObjectDeleted()
-        if r.ok:
-            return r.json()
-        else:
-            raise StorageError(f'{r.status_code} {r.text}')
+        with Cache(CACHE_DIR) as file_cache:
+            response = file_cache.get(url, None)
```

Comment:
It feels like there's an issue here: what if the indexer gets triggered, caches the list of files & object timestamps, then the object gets a file added and the indexer is triggered again, all within the FILES_EXPIRE_SECONDS timeframe? Won't the indexer ignore the new datastream, because it thinks it already knows the list of files and object timestamps?